### ISSUE TITLE
Tools/MapExtractor

### ISSUE THREAD

**Issue by MobiusDevelopment** [dufernst/LegionCore-7.3.5#76]
Maybe could update Tools/MapExtractor from TrinityCore 3.7.5
https://github.com/TrinityCore/TrinityCore/tree/7f2b7dc9c2165d2608742473a931f55b1c1a753a/src/tools
or
https://github.com/AshamaneProject/AshamaneCore/tree/legion/src/tools

**Comment by dufernst** [comment]
Tried that, but did not seem compatible with this core.

**Comment by MobiusDevelopment** [comment]
Strange I used https://github.com/MobiusDevelopment/MobiusCore extracted files and they seem to work.

**Comment by dufernst** [comment]
@MobiusDevelopment yes, they do seem to work, but phasing in many areas (in particular new Dalaran) and pathing in other areas is broken. So while everything loads I am not confident enough in manually extracted files to use those until I (or someone else) has analyzed what the differences are between those provided in the original leak and those resulting from generation.

**Comment by ReyDonovan** [comment]
more details can be seen in the pandaria code they ported the legion geodata there

**Comment by Axdrell** [comment]
I have been looking into the tools code to try and figure out why is not working. 

Questions:
1. Do you know if the tools provided with the source originally worked with the 7.3.5 (26972) client? I compared some of the Datastores files (mapping for the DBCclient files) and files under the map_extractor forlder and they are similar to the ones provided by the trinitycore legion release. So more investigation is needed. When I run the map_extractor.exe app, it is getting stuck in opening map.db2 due to either 1) not being able to find the file or 2) not being able to parse the file - hard to know for sure because the error generated from system.cpp is too broad (either file not found or bad data). 

Any information you can provide about what you suspect can be the problem will be helpful to me. I will share the fix when I have it figured out and tested.

thanks

Axdrell

**Comment by normalzero** [comment]
> 

I have found the tools dbc2 structure is not fit to 26972. The build is fit to 20365. So the solution is simple.
1. If you have an old version tool near 735 like 7.0 or 6.3 you can give a try
2. Or reverse all client

**Comment by normalzero** [comment]
> @MobiusDevelopment yes, they do seem to work, but phasing in many areas (in particular new Dalaran) and pathing in other areas is broken. So while everything loads I am not confident enough in manually extracted files to use those until I (or someone else) has analyzed what the differences are between those provided in the original leak and those resulting from generation.

The read struct must be same with write struct. So you should keep them same. And it should fit to origin logic too.

**Comment by normalzero** [comment]
> Maybe could update Tools/MapExtractor from TrinityCore 7.3.5 https://github.com/TrinityCore/TrinityCore/tree/7f2b7dc9c2165d2608742473a931f55b1c1a753a/src/tools or https://github.com/AshamaneProject/AshamaneCore/tree/legion/src/tools

These are same with 335. But in this core vmo has a "BSPX" section

**Comment by normalzero** [comment]
> more details can be seen in the pandaria code they ported the legion geodata there [ReyDonovan/LegacyMoP@5a44ee0](https://github.com/ReyDonovan/LegacyMoP/commit/5a44ee01e5425f8ed39f73579100113912334bd3)

The tools don't update just collision check

**Comment by frostfire11** [comment]
@normalzero Do you have a solution for this?

**Comment by qyh214** [comment]
Looking forward to the updates in this section, but it may be very difficult if there are any special changes
